File: metrics/test/data/test_cognitive_complexity/test_recursion/data_recursion.java
public class data_recursion {
    public Main() {
    }

    public void recurse(Integer i) {
        if (recurse(i + 1)) {
            recurse(i);
        } else {
            recurse(i+3);
        }
        System.out.println(recurse(2) + recurse(3));
        return recurse(i + 5);
    }

    public void Print(String s) {
        System.out.println(s);
    }

    public void HelloWorld() {
        Print("Hello World");
    }
}

File: metrics/test/data/test_cognitive_complexity/test_recursion/data_recursion.cs
public class TestClass
{
    public TestClass() { }

    public void recurse(int i)
    {
        recurse(i + 1);

        recurse(i);
        recurse(1);

        Console.WriteLine(recurse(i + 2) + recurse(i + 3));

        return (recurse(5));
    }

    private void print(string s)
    {
        Console.WriteLine(s);
    }

    public void HelloWorld()
    {
        print("Hello world");
    }
}